### PR TITLE
ref(project-config): Disallow non-full config requests

### PR DESCRIPTION
**Pull request by iambriccardo**
This PR disallows non-full config requests in the project configuration endpoint. It also removes the `full_config` booleans throughout the code, to simplify it.

Closes: https://github.com/getsentry/team-ingest/issues/84

### CODE REVIEW

File: src/sentry/relay/config/__init__.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def get_exposed_features(project: Project) -> Sequence[str]:
 
 
 def get_public_key_configs(
-    project: Project, full_config: bool, project_keys: Iterable[ProjectKey] | None = None
```
Comment (iambriccardo) [inline src/sentry/relay/config/__init__.py:115]: `project` was unused.

File: src/sentry/api/endpoints/relay/project_configs.py [hunk 2]
```diff
@@ -43,10 +44,15 @@ def post(self, request: Request):
...
-            return Response("Relay unauthorized for full config information", 403)
+        if not request.relay_request_data.get("fullConfig"):
+            # We capture a message to make it apparent to self-hosted users that now it's only possible to query
+            # full project configs.
+            sentry_sdk.capture_message(
+                "Requesting the non-full configuration is not possible anymore"
+            )
+            return Response(
+                "Requesting the non-full configuration is not possible anymore", status=400
+            )
```
Comment (jjbayer) [inline src/sentry/api/endpoints/relay/project_configs.py:55]: We still need the `is_internal` check, but that also means we can simplify and ignore the parameter entirely IMO. The 403 should be signal enough.
```suggestion
        if not relay.is_internal:
            return Response("Relay unauthorized for config information", 403)
```

**Review by jjbayer — CHANGES_REQUESTED** [review]
Looks good! We do need the `is_internal` check though.

File: src/sentry/api/endpoints/relay/project_configs.py (hunk 2 above)
Comment (iambriccardo) [reply src/sentry/api/endpoints/relay/project_configs.py:55]: Yeah I was wondering if we wanted that, I thought we wanted to allow external relays to get the config. My bad for misunderstanding

File: tests/sentry/api/endpoints/test_relay_projectconfigs.py [hunk 3]
```diff
@@ -169,7 +164,7 @@ def test_internal_relays_should_receive_full_configs(
 
 
 @django_db_all
-def test_relays_dyamic_sampling(client, call_endpoint, default_project, dyn_sampling_data):
```
Comment (iambriccardo) [inline tests/sentry/api/endpoints/test_relay_projectconfigs.py:172]: Removed unused fixtures.

File: tests/sentry/api/endpoints/test_relay_projectconfigs.py [hunk 4]
```diff
@@ -190,57 +185,60 @@ def test_relays_dyamic_sampling(client, call_endpoint, default_project, dyn_samp
...
 ):
     relay.is_internal = False
     relay.save()
 
-    result, status_code = call_endpoint(full_config=False)
+    result, status_code = call_endpoint()
 
     assert status_code < 400
 
+    # Sweeping assertion that we do not have any snake_case in that config.
```
Comment (iambriccardo) [inline tests/sentry/api/endpoints/test_relay_projectconfigs.py:232]: Copied all new lines of this test from `test_internal_relays_should_receive_full_configs` and just modified `assert safe.get_path(cfg, "config", "trustedRelays") != []`.

File: tests/sentry/api/endpoints/test_relay_projectconfigs.py [hunk 5]
```diff
@@ -281,21 +280,6 @@ def projectconfig_cache_set(monkeypatch):
     return calls
 
 
-@django_db_all
-def test_relay_projectconfig_cache_minimal_config(
```
Comment (iambriccardo) [inline tests/sentry/api/endpoints/test_relay_projectconfigs.py:285]: Not needed anymore.

File: src/sentry/api/endpoints/relay/project_configs.py [hunk 6]
```diff
@@ -43,10 +44,18 @@ def post(self, request: Request):
...
-        full_config_requested = request.relay_request_data.get("fullConfig")
+        if not request.relay_request_data.get("fullConfig", True):
+            # We capture a message to make it apparent to self-hosted users that now it's only possible to query
+            # full project configs.
+            sentry_sdk.capture_message(
+                "Requesting the non-full configuration is not possible anymore"
+            )
+            return Response(
+                "Requesting the non-full configuration is not possible anymore", status=400
+            )
```
Comment (jjbayer) [inline src/sentry/api/endpoints/relay/project_configs.py:55]: As discussed, let's just remove this.

File: tests/sentry/api/endpoints/test_relay_projectconfigs.py [hunk 7]
```diff
@@ -190,57 +185,60 @@ def test_relays_dyamic_sampling(client, call_endpoint, default_project, dyn_samp
...
-    result, status_code = call_endpoint(full_config=None)
+    result, status_code = call_endpoint()
 
     assert status_code < 400
 
     cfg = safe.get_path(result, "configs", str(default_project.id))
-    assert safe.get_path(cfg, "config", "filterSettings") is None
-    assert safe.get_path(cfg, "config", "groupingConfig") is None
+    assert safe.get_path(cfg, "config", "filterSettings") is not None
+    assert safe.get_path(cfg, "config", "groupingConfig") is not None
```
Comment (jjbayer) [inline tests/sentry/api/endpoints/test_relay_projectconfigs.py:218]: status code should be 403 here, right?

File: tests/sentry/api/endpoints/test_relay_projectconfigs.py [hunk 8]
```diff
@@ -190,57 +185,60 @@ def test_relays_dyamic_sampling(client, call_endpoint, default_project, dyn_samp
...
 
 
 @django_db_all
-def test_trusted_external_relays_should_receive_minimal_configs(
+def test_trusted_external_relays_should_receive_full_configs(
```
Comment (jjbayer) [inline tests/sentry/api/endpoints/test_relay_projectconfigs.py:222]: A lot of these tests can probably now be de-duplicated, could you look into removing some of them?

Comment (iambriccardo) [reply tests/sentry/api/endpoints/test_relay_projectconfigs.py:222]: I was considering doing that, wanted first to get through a review.

**Review by jjbayer — APPROVED** [review]
Let's merge https://github.com/getsentry/self-hosted/pull/3144 before this one.

**Pull request merged**